### PR TITLE
Support Scala3

### PR DESCRIPTION
**Pull request by d2a4u**
- Remove Scanamo module as it is blocking Scala3 support
- Made some deprecated classes private

### CODE REVIEW

**Comment by marko-asplund** [comment]
The logs appear to have expired, so I'm unable to see the error anymore, but here's the error I'm getting with commit 1c2cdb652bd9b5dbcedd71c9917bce8dd13bc11d:
[meteor-pr301-fail.txt](https://github.com/d2a4u/meteor/files/12327456/meteor-pr301-fail.txt)

**Comment by marko-asplund** [comment]
Scala [v3.3.1](https://github.com/lampepfl/dotty/releases/tag/3.3.1) was released on 2023-09-08.
Any chance of someone looking at this issue again?
Please also check out https://github.com/d2a4u/meteor/pull/343.

File: project/plugins.sbt [hunk 1]
```diff
@@ -1,3 +1,4 @@
 addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "2.4.2")
 addSbtPlugin("ch.epfl.scala" % "sbt-release-early" % "2.1.1")
 addSbtPlugin("io.github.davidgregory084" % "sbt-tpolecat" % "0.1.22")
+addSbtPlugin("ch.epfl.scala" % "sbt-scala3-migrate" % "0.5.1")
```
Comment (marko-asplund) [inline project/plugins.sbt:4]: Is this needed?

**Comment by marko-asplund** [comment]
Thanks for getting back @d2a4u ! 🙇

I can try to review, but I'm not familiar with the codebase from before, and as a result may end up asking lots of questions.

Based on a quick look, many of the changes appear to be Scala 2 compatible e.g. trivial syntax changes or Meteor internal API refactoring. Would it be possible to extract these changes to a PR separate from Scala 2 incompatible changes needed to make the leap to Scala 3? 🤔 Another way would be to split into Scala 2 compatible internal API refactorings and the rest. This would make it easier to reason about the review the changes.

Would the resulting Scala 3 libraries be compatible with Scala versions >= 3.0 and < 3.3?

File: build.sbt [hunk 2]
```diff
@@ -22,13 +21,13 @@ lazy val testDependencies = Seq(
 
 lazy val ItTest = config("it").extend(Test)
 
-lazy val scala213 = "2.13.11"
-lazy val scala212 = "2.12.18"
+lazy val scala3 = "3.3.1"
+lazy val scala213 = "2.13.12"
```
Comment (marko-asplund) [inline build.sbt:25]: Maybe good to call out in the PR that the project won't be cross compiled for Scala v2.12.

File: build.sbt [hunk 3]
```diff
@@ -67,10 +63,19 @@ lazy val root = project
...
-    scanamo % "compile->compile;test->test",
     dynosaur % "compile->compile;test->test"
   )
-  .aggregate(awssdk, scanamo, dynosaur)
+  .settings(
+    Compile / scalacOptions ++= {
+      CrossVersion.partialVersion(scalaVersion.value) match {
+        case Some((3, _)) =>
+          Seq("-Ykind-projector:underscores")
+        case Some((2, 12 | 13)) =>
```
Comment (marko-asplund) [inline build.sbt:73]: Is `12` needed if Meteor isn't cross compiled for 2.12 anymore? 🤔

File: build.sbt [hunk 4]
```diff
@@ -108,20 +113,3 @@ lazy val dynosaur = project
     commonSettings
   ).dependsOn(awssdk)
 
-lazy val scanamo = project
```
Comment (marko-asplund) [inline build.sbt:111]: Looks like this PR removes Scanamo codec support. Should the docs be updated accordingly? 🤔

**Review by marko-asplund — APPROVED** [review]
Left some comments, but looks good to me 👍

**Comment by marko-asplund** [comment]
@d2a4u Sorry, for the delay!
Finally got around doing the review.
The PR looks good to me, so I approved, but left some minor comments.
Thanks again for your work on Meteor in general as well as adding Scala 3 support! 🥇 🙇

**Comment by marko-asplund** [comment]
Hi again! 👋 How are things looking with finalising this PR @d2a4u ? 😄 Are there some impediments?

**Comment by d2a4u** [comment]
> Hi again! 👋 How are things looking with finalising this PR @d2a4u ? 😄 Are there some impediments?

Sorry for the delay, I had some issues with sonatype which now have been resolved.

**Comment by marko-asplund** [comment]
Great to see a new version of meteor released with Scala 3 support! 🥳 Thanks @d2a4u ! 🙇

**Pull request merged**